pkg/server/proto/server.proto

```diff
@@ -1143,6 +1159,47 @@ message Task {
   opaqueany.Any state = 6;
 }
 
+message UpsertTaskRequest {
+  // The Task to upsert. If the id in the Task message is empty, then this will
+  // be an insert in the db. Otherwise if an id was requested, this will act
+  // as an update. If the ID isn't found, it will be an error.
+  Task task = 1;
+}
+
+message UpsertTaskResponse {
+  // The resulting Task that was Upserted.
+  Task task = 1;
+}
+
+message GetTaskRequest {
+  // Can be oneof Task id or by the Run job id.
+  Ref.Task ref = 1;
+}
+
+message GetTaskResponse {
```

Comment:
On a similar note, I wonder if it makes sense to add these as "computed" fields on the `Task` message so that the state system can generate them. This is similar to how "job" has computed fields.